src/smt/smt_engine.cpp

```diff
@@ -1230,7 +1205,7 @@ Result SmtEngine::checkSatisfiability(const vector<Node>& assumptions,
     Result::UnknownExplanation why = d_resourceManager->outOfResources()
                                          ? Result::RESOURCEOUT
                                          : Result::TIMEOUT;
-    return Result(Result::SAT_UNKNOWN, why, d_filename);
+    return Result(Result::SAT_UNKNOWN, why, d_state->getFilename());
```

Comment:
Don't we also need to inform our state about this result?

Reply:
No. SmtEngineState::notifyCheckSatResult() is for cases where the result was computed and the solver should continue as normal.  I would recommend a separate callback for this case, e.g. SmtEngineState::notifyInterupt(), which however would do nothing at the moment (which is why I haven't added it).  I'll add a comment about this.

Regardless this is making it much more clear to me what we would need to do to implement safe interupts.

Reply:
Hmm, I think that this breaks doing `(get-info :reason-unknown)` after a timeout, no? Maybe this is a bit out of scope for this PR though. We probably need a separate state `INVALID` until our interrupts are safe.

Reply:
Sure, note this PR didnt change this behavior.